x/nexus/exported/types.go

```diff
@@ -17,7 +19,15 @@ import (
 	"github.com/axelarnetwork/utils/slices"
 )
 
-//go:generate moq -out ./mock/types.go -pkg mock . MaintainerState
+//go:generate moq -out ./mock/types.go -pkg mock . MaintainerState LockableCoin
+
+// LockableCoin defines a coin that can be locked and unlocked
+type LockableCoin interface {
+	GetCoin() sdk.Coin
+	GetOriginalCoin(ctx sdk.Context) sdk.Coin
```

Comment:
```suggestion
	GetAsset() sdk.Coin
	GetCoin(ctx sdk.Context) sdk.Coin
```
to differentiate nexus registered asset denomination from actual coins